scipy/stats/_distn_infrastructure.py

```diff
@@ -2164,16 +2176,39 @@ def restore(args, theta):
 
             def func(theta, x):
                 newtheta = restore(args[:], theta)
-                return self._penalized_nnlf(newtheta, x)
+                return objective(newtheta, x)
 
         return x0, func, restore, args
 
+    def _moment_error_norm(self, theta, x):
```

Comment:
This is where the magic happens. All the rest are minor tweaks to work method of moments into the existing structure. The first two lines were copied from MLE without questioning them.